app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/AddReviewFragment.kt

```diff
@@ -43,32 +44,31 @@ class AddReviewActivity : AppCompatActivity() {
     private lateinit var reviewIndicationTitle: TextView
     private lateinit var scoreTextView: TextView
     private lateinit var doneButton: Button
-    private var reviewableId: String? = null
-
-    val viewModel: AddReviewViewModel by viewModels()
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        setContentView(R.layout.add_review_layout)
+    private val viewModel: AddReviewViewModel by viewModels()
+    private val activityViewModel by activityViewModels<ReviewViewModel>()
 
-        reviewableId = intent.getStringExtra(EXTRA_ITEM_REVIEWED)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
 
-        doneButton = findViewById(R.id.doneButton)
-        ratingBar = findViewById(R.id.reviewRatingBar)
-        comment = findViewById(R.id.addReviewComment)
-        title = findViewById(R.id.addReviewTitle)
-        reviewIndicationTitle = findViewById(R.id.reviewTitle)
-        scoreTextView = findViewById(R.id.overallScoreTextView)
+        doneButton = view.findViewById(R.id.doneButton)
+        ratingBar = view.findViewById(R.id.reviewRatingBar)
+        comment = view.findViewById(R.id.addReviewComment)
+        title = view.findViewById(R.id.addReviewTitle)
+        reviewIndicationTitle = view.findViewById(R.id.reviewTitle)
+        scoreTextView = view.findViewById(R.id.overallScoreTextView)
 
-        viewModel.rating.observe(this) { rating ->
+        viewModel.rating.observe(viewLifecycleOwner) { rating ->
             scoreTextView.text =
                 getString(
                     R.string.overall_score_review,
                     rating?.toString() ?: NO_GRADE_MESSAGE
                 )
         }
 
-        reviewIndicationTitle.text = getString(R.string.title_review, reviewableId)
+        activityViewModel.getReviewable().observe(viewLifecycleOwner){
+            reviewIndicationTitle.text = getString(R.string.title_review, it!!.id)
```

Comment:
I'd prefer it if you avoid the !!. If you could use a null check or a ?.apply, it would be better I guess

Reply:
I thought it was the same, my bad :(